pretyping/coercion.ml

```diff
@@ -622,6 +622,13 @@ let lookup_reversible_path_to_common_point env sigma ~src_expected ~src_inferred
   in
     aux r
 
+type hook = env -> evar_map -> flags:Evarconv.unify_flags -> constr ->
+  inferred:types -> expected:types -> (evar_map * constr) option
+let hook = Summary.ref ~name:"coercion_hook"
```

Comment:
This is not correct. You cannot store functions in a Summary (more generally, all summary objects must be marshallable). Instead you should add an indirection based on a static table associating functions to strings and storing instead a string in the summary.

Reply:
It could also be a local summary ref (`Summary.ref ~local:true`) depending on what the desired behaviour is

Reply:
Given the use case I don't think we want a local reference.

Reply:
Indeed thanks for checking this will work on backtracking the plugin setting the hook.

Reply:
Thanks, you're right there is probably no other way than handling the Summary in each plugin. Added a comment to warn users.

Reply:
@proux01 how would your approach work? Plugins can't coordinate among themselves I think.

Reply:
The initial idea was to offer `get_hook` and `set_hook` so that plugins can `set_hook` to a function that would run their own hook + `get_hook` in whatever order they prefer. I agree this is not perfectly robust to backtracking on plugin loading which could lead to hooks being called multiple times / in different orders.
I thus changed the interface to a list of hooks with some local summary. This should be better although this probably still doesn't work with async proofs (but I don't know enough about async to tell or even test). I hope this solves the issue.